app/test/unit/git/squash-test.ts

```diff
@@ -34,18 +45,269 @@ describe('git/cherry-pick', () => {
     expect(squashed.summary).toBe('Test Summary')
     expect(squashed.body).toBe('Test Body\n')
     expect(log.length).toBe(2)
+
+    // verify squashed commit contains changes from squashed commits
+    const squashedFiles = await getChangedFiles(repository, squashed.sha)
+    const squashedFilePaths = squashedFiles.map(f => f.path).join(' ')
+    expect(squashedFilePaths.includes('first.md')).toBeTrue()
+    expect(squashedFilePaths.includes('second.md')).toBeTrue()
+  })
+
+  it('squashes multiple commit onto one (non-conflicting)', async () => {
+    const firstCommit = await makeSquashCommit(repository, 'first')
+    const secondCommit = await makeSquashCommit(repository, 'second')
+    const thirdCommit = await makeSquashCommit(repository, 'third')
+    const fourthCommit = await makeSquashCommit(repository, 'fourth')
+
+    const result = await squash(
+      repository,
+      [secondCommit, thirdCommit, fourthCommit],
+      firstCommit,
+      initialCommit.sha,
+      'Test Summary\n\nTest Body'
+    )
+
+    expect(result).toBe(RebaseResult.CompletedWithoutError)
+
+    const log = await getCommits(repository, 'HEAD', 5)
+    const squashed = log[0]
+    expect(squashed.summary).toBe('Test Summary')
+    expect(squashed.body).toBe('Test Body\n')
```

Comment:
I just noticed the extra `\n`. Is that correct? 🤔

Reply:
Yes - I am not sure why.. and it totally perplexed me at first but if you look at the commit unit tests we do that there too.

Reply:
It also was curious to me that we have to double new line between the summary and body.. but seems that is what git expects.

Reply:
> but seems that is what git expects

Yeah, I was aware of that, but for some reason I expected our code to drop both new lines when it parses the body and summary 🤔